src/machine/stack.rs

```diff
@@ -245,7 +254,7 @@ impl Stack {
     #[inline(always)]
     pub(crate) fn index_and_frame(&self, e: usize) -> &AndFrame {
         unsafe {
-            let ptr = self.buf.base as usize + e;
+            let ptr = self.buf.get_unchecked(e);
```

Comment:
Lot's of unsoundness in this file, but from your comment in `fn index()` I guess you are aware. This was already unsound before your changes though.